dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetMetadataTab.java

```diff
@@ -80,6 +81,15 @@ public boolean canUpdateDataset() {
         return permissionsWrapper.canUpdateDataset(dvRequestService.getDataverseRequest(), this.dataset);
     }
 
+    /**
+     * Updates the dataset lock state and refreshes button in metadataTab xhtml.
+     */
+    public boolean updateDatasetLockState(boolean isDatasetLocked) {
+        this.isDatasetLocked = isDatasetLocked;
+        PrimeFaces.current().ajax().update("datasetForm:tabView:editMetadataButton");
```

Comment:
Not sure if update here is elegant. We depend here very tightly on component names in xhtml. It is easy to miss this when for example we want to change id's of some component. It is defined in xhtml and some handling is in java classes.
I am wondering if we could change update attribute of `<p:commandButton id="refreshButton" process="@this" widgetVar="refreshButton" update="...` on `dataset.xhtml`.

Reply:
I've made a temporary button to see if it's working, so it should be working.